lib/pdk/cli/validate.rb

```diff
@@ -19,6 +19,11 @@ module PDK::CLI
         exit 0
       end
 
+      if PDK::Util.module_root.nil?
+        PDK.logger.error(_('pdk validate must be run inside a module'))
+        exit 1
+      end
```

Comment:
1) Please extract that to `PDK::CLI::Util.check_in_module!` and apply liberally to all the commands that should run in a module:

* pdk new class
* pdk test unit
* pdk validate

2) add `(no metadata.json found)` to ensure users know what we're missing.